packages/cosmic-swingset/lib/ag-solo/vats/lib-wallet.js

```diff
@@ -261,48 +226,50 @@ export async function makeWallet(
     return offerOk;
   }
 
-  // === API
-
-  async function addPurse(purse, assayId) {
-    const purseNameP = E(purse).getName();
-    const assayP = E(purse).getAssay();
-    const labelP = E(assayP).getLabel();
+  const getLocalUnitOps = assay =>
+    Promise.all([
+      E(assay).getLabel(),
+      E(assay).getExtentOps(),
+    ]).then(([label, { name, extentOpsArgs = [] }]) =>
+      makeUnitOps(label, name, extentOpsArgs),
+    );
 
-    const [purseName, assay, { allegedName }] = await Promise.all([
-      purseNameP,
-      assayP,
-      labelP,
-    ]);
+  // === API
 
-    // =====================
-    // === AWAITING TURN ===
-    // =====================
+  async function addAssay(assayPetname, regKey, assay) {
+    petnameToAssay.init(assayPetname, assay);
+    assayToPetname.init(assay, assayPetname);
+    regKeyToAssayPetname.init(regKey, assayPetname);
+    assayPetnameToRegKey.init(assayPetname, regKey);
+    petnameToUnitOps.init(assayPetname, await getLocalUnitOps(assay));
+  }
 
-    insist(!nameToPurse.has(purseName))`Purse name already used in wallet.`;
 
-    // =====================
-    // === AWAITING TURN ===
-    // =====================
+  async function makeEmptyPurse(assayPetname, pursePetname, memo = 'purse') {
+    insist(
+      !petnameToPurse.has(pursePetname),
+    )`Purse name already used in wallet.`;
+    const assay = petnameToAssay.get(assayPetname);
 
     // IMPORTANT: once wrapped, the original purse should never
     // be used otherwise the UI state will be out of sync.
+    const _ = await E(assay).makeEmptyPurse(memo);
```

Comment:
Hey Kate, I'm not a big fan of using underscore for a purse. We typically use underscore for variables we want to ignore.

https://eslint.org/docs/rules/no-unused-vars

Reply:
Hmm, I was intending to indicate strongly that this purse should never be used (the observable purse should be used instead) by calling it _.  Is there another name that would indicate the same thing?

Reply:
Just changed this to 'doNotUse'